src/app/auth/auth.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { SignupCredentials } from '../signup_credentials';
+import { SigninCredentials } from '../signin_credentials';
+import { BehaviorSubject, tap, Observable } from 'rxjs';
+
+interface SignupResponse {
+  username: string;
+}
+
+interface SignedinResponse {
+  authenticated: boolean;
+  username: string;
+}
+
+interface SigninResponse {
+  username: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  rootUrl: string = 'https://api.angular-email.com';
+  signedin$: BehaviorSubject<any> = new BehaviorSubject(null);
+  username: string = '';
+
+  constructor(private http: HttpClient) {}
+
+  usernameAvailable(username: string): Observable<{ available: boolean }> {
+    return this.http.post<{ available: boolean }>(
+      this.rootUrl + '/auth/username',
```

Comment:
Please refactor so that all the api urls are fetched from a common service. In this case if we need to change the endpoints we will only need to change in one file instead of searching and changing through the whole codebase.